RBQFetchedResultsController/Source/RBQFetchedResultsController.m

```diff
@@ -915,12 +915,12 @@ - (void)applyDerivedChangesToCache:(RBQDerivedChangesObject *)derivedChanges
                 RBQSectionCacheObject *section =
                 [RBQSectionCacheObject objectInRealm:state.cacheRealm
                                        forPrimaryKey:objectChange.updatedCacheObject.sectionKeyPathValue];
-                
-#ifdef DEBUG
-                NSAssert(objectChange.updatedIndexpath.row <= section.objects.count, @"Attemting to insert at index beyond bounds!");
-#endif
+                NSInteger insertIndex = objectChange.updatedIndexpath.row;
+                if (insertIndex >= section.objects.count) {
+                    insertIndex = section.objects.count - 1;
+                }
```

Comment:
So, then is the problem in the logic that shouldn't be allowing this to begin with? Obviously it's happening to some people, so what might be causing it in the first place.

I do agree that it's better to fix the cause of the problem rather than trying to handle when it happens here though.